aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -345,35 +345,36 @@ module aptos_token::token {
                 };
                 // update the token largest property_version
                 direct_deposit(token_owner, new_token);
-                update_token_property_internal(token_owner, new_token_id, keys, values, types);
+                update_token_property_internal(token_owner, token_id, new_token_id, keys, values, types);
                 i = i + 1;
             };
             token_data.largest_property_version = largest_property_version + token.amount;
             // burn the orignial property_version 0 token after mutation
             let Token {id: _, amount: _, token_properties: _} = token;
         } else {
             assert!(amount == 1, ENON_ZERO_PROPERTY_VERSION_ONLY_ONE_INSTANCE);
-            update_token_property_internal(token_owner, token_id, keys, values, types);
+            update_token_property_internal(token_owner, token_id, token_id, keys, values, types);
         };
     }
 
     fun update_token_property_internal(
         token_owner: address,
-        token_id: TokenId,
+        old_token_id: TokenId,
+        new_token_id: TokenId,
         keys: vector<String>,
         values: vector<vector<u8>>,
         types: vector<String>,
     ) acquires TokenStore {
         let tokens = &mut borrow_global_mut<TokenStore>(token_owner).tokens;
-        assert!(table::contains(tokens, token_id), ENO_TOKEN_IN_TOKEN_STORE);
-        let value = &mut table::borrow_mut(tokens, token_id).token_properties;
+        assert!(table::contains(tokens, new_token_id), ENO_TOKEN_IN_TOKEN_STORE);
+        let value = &mut table::borrow_mut(tokens, new_token_id).token_properties;
 
         property_map::update_property_map(value, keys, values, types);
         event::emit_event<MutateTokenPropertyMapEvent>(
             &mut borrow_global_mut<TokenStore>(token_owner).mutate_token_property_events,
             MutateTokenPropertyMapEvent {
-                old_id: token_id,
-                new_id: token_id,
+                old_id: old_new_token_id,
```

Comment:
the old_id will always be the one with property_version 0.
can you make this `old_id: create_token_id(token_id.token_data_id, 0);`?  and remove all other changes?

Reply:
From L354 in the `else` block I think the `old_id` can be a `TokenId` with `property_version` > 0 ? If `token_id.property_version == 0`, the code in `if` block will be executed and if `token_id.property_version > 0` the code in `else` block` will be executed, we actually need a method to know whether the original token was changed or new tokens were created ...

Reply:
there's a typo in this line, just fixed and rebased from `origin/main`, resolved conflicts

Reply:
This is fixed https://github.com/aptos-labs/aptos-core/pull/4118/files as well. But all credits to you for identifying this issue. Thanks !